File: TestBankApp/app/src/main/java/com/example/bankapp/domain/UserContract.java
package com.example.bankapp.domain;

import com.example.bankapp.helper.BaseCallback;
import com.example.bankapp.data.remote.model.user.UserAccountModel;

public class UserContract {
    public interface IRepository{
        void login(String userName, String password, BaseCallback<UserAccountModel> result);
    }
}

File: BankApp/app/src/main/java/br/com/vinicius/bankapp/domain/UserContract.kt
package br.com.vinicius.bankapp.domain

import br.com.vinicius.bankapp.internal.BaseCallback

class UserContract {

    interface IUser{
        val username: String
        val password: String

        fun isValid(): Boolean

        fun startLogin(listener: BaseCallback<User>)
    }

    interface IRepository {
        fun startLogin(username: String, password: String, onResult: BaseCallback<User>)
    }
}